### PR TITLE
[SIS-128] Clean up activity card search

### LINKED ISSUE #128: [SIS-128] Clean up activity card search
- [ ] GET FEEDBACK FROM USER TESTING, but tags should bring up search search results. Doesn't make sense to search by title
- [ ] Instead of search results showing up in a dropdown that requires you to focus on the textInput/searchbar, it should show up in a component underneath

### PR DESCRIPTION
**Pull request by BakerWJ**
# Description

Implemented request caching with React Query to improve performance of searching for tickets. Also implemented tag filtering.

Ticket #128 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Style (formatting, missing semi colons, etc; no production code change)
- [x] Refactor (refactoring production code, eg. renaming a variable)

# Checklist:

- [x] My PR title is formatted as [SIS-\<issue number\>] \<issue name\> (eg. [SIS-010] Create UI button)
- [x] I have linked my issue/ticket on the project board to this PR
- [ ] I have sent my PR to #team-sistema-prs on Slack and requested two people to review
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by BakerWJ** [comment]
One last thing that I haven't been able to figure out is having the menu always open, even if we aren't typing. Maybe we make this decision after user testing?

File: src/editor/components/SearchResults.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import axios from 'axios';
```
Comment (ramyzhang) [inline src/editor/components/SearchResults.js:1]: nice to isolate these calls to the service file

File: src/editor/screens/AddActivityCard.js
Comment (ramyzhang) [inline src/editor/screens/AddActivityCard.js:1]: missing the "no matching activity cards found" response here, but we can merge this for now

**Review by ramyzhang — APPROVED** [review]
great!! merging

**Pull request merged**